apps/comps/hooks/useClaim.ts

```diff
@@ -0,0 +1,203 @@
+        chainId: publicConfig.blockchain.chain.id,
+      };
+    });
+  }, [allClaims, address]);
+
+  const { data: claimedStatus, queryKey: claimedStatusQueryKey } =
+    useSafeReadContracts({
+      contracts,
+      query: {
+        enabled: contracts.length > 0,
+      },
+    });
+
+  // Filter out claims that have already been claimed
+  const claims = useMemo(() => {
+    if (!allClaims || !claimedStatus) return [];
+
+    return allClaims.filter((_, index) => {
+      const result = claimedStatus[index];
+      // If the result is successful and the value is false, the claim hasn't been claimed yet
+      return result?.status === "success" && result.result === false;
+    });
+  }, [allClaims, claimedStatus]);
+
+  const totalClaimable = useMemo(() => {
+    if (!claims || claims.length === 0) return 0n;
+    return claims.reduce((acc, c) => acc + BigInt(c.amount), 0n);
+  }, [claims]);
+
+  const refetchQueries = async (txHash: `0x${string}`) => {
+    const transactionReceipt = await waitForTransactionReceipt(
+      clientConfig as any,
+      {
+        hash: txHash,
+        pollingInterval: 1000,
+      },
+    );
+
+    if (transactionReceipt.status === "success") {
+      setIsConfirmed(true);
+      setIsConfirming(false);
+      queryClient.invalidateQueries({
+        queryKey: tanstackClient.rewards.getClaimData.key(),
+      });
+      queryClient.invalidateQueries({
+        queryKey: claimedStatusQueryKey,
+      });
+    }
+  };
+
+  // Execute function that simulates before claiming
+  const claim = useCallback(
+    async (item?: ClaimItem) => {
+      try {
+        const target = item ?? claims?.[0];
+        if (!target) throw new Error("No claimable rewards");
+
+        // Reset the write contract hook
+        reset();
+
+        const root = target.merkleRoot as Hex;
+        const amount = BigInt(target.amount);
+        const proof = target.proof as Hex[];
+
+        // First simulate the transaction
+        const simulationResult = await simulateContract(config as any, {
+          address: publicConfig.blockchain
+            .rewardAllocationContractAddress as Hex,
+          abi: RewardAllocationAbi,
+          functionName: "claim",
+          args: [root, amount, proof],
+          account: address,
+        });
+
+        // If simulation succeeds, execute the actual transaction
+        writeContract(simulationResult.request, {
+          onSuccess: (txHash) => {
+            refetchQueries(txHash);
+          },
+          onError: (error) => {
+            setIsConfirming(false);
+            throw new Error(error.message);
+          },
+        });
+        setIsConfirming(true);
+      } catch (simulationError) {
+        setIsConfirming(false);
+        // Re-throw simulation errors with a clear message
+        throw new Error(
+          `Transaction simulation failed: ${simulationError instanceof Error ? simulationError.message : "Unknown error"}`,
+        );
```

Comment:
Seems like you're throwing an uncaught error here (unless the caller wraps it in a try/catch which is unlikely in a React component).